shared/constraints.py

```diff
@@ -99,3 +98,17 @@ def combine_linear(linear_callables: Sequence[LinearCallable]) -> Tuple[np.ndarr
     A = np.array([c.a for c in linear_callables], dtype=np.float64)
     b = np.array([c.b for c in linear_callables], dtype=np.float64)
     return A, b
+
+
+def combine_linear_constraints(constraints: List[LinearConstraint]) -> Tuple[np.ndarray, np.ndarray]:
+    """Combine attributes of Constraints into matrix A and vector b.
+
+    Args:
+        constraints (List[LinearConstraints]): List of linear constraints.
+
+    Returns:
+        Tuple[np.ndarray, np.ndarray]: A and b.
+    """
+    A = np.array([const.c.a for const in constraints], dtype=np.float64)
```

Comment:
(minor) man könnte hier combine_linear verwendn um bissl Code-Duplizierung zu vermeiden. Aber, dann müsstn wir ein neues Array erzeugen aus den LinearCallables, also kann man auch for the sake of performance argumentieren, wahrschl. aber neglectable.